doc/spec_TransferElement.dox

```diff
@@ -213,9 +232,49 @@ This documnent is currently still **INCOMPLETE**!
     - \anchor transferElement_B_11_7_1 11.7.1 *either* do not guarantee that consecutive calls to TransferElement::getVersionNumber() always return a VersionNumber that is equal or greater than the one from the previous call \ref transferElement_comment_B_11_7_1 "(*)"
     - \anchor transferElement_B_11_7_2 11.7.2 *or* discard values with lower version numbers \ref transferElement_comment_B_11_7_2 "(*)"
 
-- 12. TransferGroup
-  - 12.1 TransferGroups are only allowed for TransferElements without AccessMode::wait_for_new_data.
-  -  [TODO]
+- 12. TransferGroup and merging transfers:
+  - 12.1 TransferElements which are put into the same TransferGroup can optionally merge their data transfer to optimise for speed and to offer a best-effort data consistency.
+    - 12.1.1 TransferGroups do not enforce data consistency, since merging transfers is optional.
+      - 12.1.1.1 If a merge happens and consistency of the data can be guaranteed in a read operation, the VersionNumber of the corresponding TransferElements is set to the same value.
+    - 12.1.2 Structure of TransferElements in TransferGroups:
+      - \anchor transferElement_B_12_1_2_1 12.1.2.1 A TransferElement, which is directly added to the TransferGroup, is called <i>high-level element</i>.
+      - \anchor transferElement_B_12_1_2_2 12.1.2.2 Decorator-like implementations use one or more <i>low-level elements</i> which perform the actual register access. All low-level elements of one TransferElement are returned by TransferElement::getHardwareAccessingElements().
+      - 12.1.2.3 Low-level elements are not decorator-like, since they directly access the device.
+      - 12.1.2.4 There can be any number of decorator-like intermediate elements. The sum of all elements on all levels of one high-level element including the low-level elements but excluding the high-level element itself is called <i>internal elements</i>. All internal elements of one TransferElement are returned by TransferElement::getInternalElements().
```

Comment:
More of a comment: This might also make sense to put it into some kind of TransferGroup glossary or detailed description, outside of this spec. This is important terminology imho and should not be buried inside the TE spec